pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java

```diff
@@ -151,7 +158,12 @@ public Record<GenericObject> process(GenericObject input, Context context) throw
 
   @Override
   public void process(TransformContext transformContext) throws Exception {
-    for (TransformStep step : steps) {
+    for (StepPredicatePair pair : steps) {
+      TransformStep step = pair.getTransformStep();
+      Predicate<TransformContext> predicate = pair.getPredicate();
+      if (predicate != null && !predicate.test(transformContext)) {
+        continue;
```

Comment:
Should reverse this condition and avoid the `continue`

Reply:
sure, I wanted to use the filter syntax in streams to make it cleaner, but it forces us to handle the checked exception in-place...